tests/framework/class-wpseo-unit-test-case.php

```diff
@@ -33,6 +33,24 @@ protected function go_to_home() {
 		$this->go_to( home_url( '/' ) );
 	}
 
+	/**
+	 * Invokes a class method, ignoring whether it is protected or private.
+	 *
+	 * @param object $instance    Class instance to invoke the method on.
+	 * @param string $method_name Method name.
+	 * @param array  $args        Optional. Method arguments. Default empty array.
+	 *
+	 * @return mixed Invoked method return value.
+	 */
+	protected function invoke_method( $instance, $method_name, $args = array() ) {
```

Comment:
I think a `ReflectionClass` will add some magic layer to the code and it pretty hard to understand. Please make the methods public by default or consider having a class method where the methods get exposed.

Reply:
I think the methods shouldn't be public because they shouldn't be part of the public interface. I thought it would make sense to still be able to easily test them. The above `invoke_method()` method would help significantly improve coverage of such protected or private methods.

If we shouldn't have `invoke_method()`, I don't see another way we can unit-test such methods without restructuring huge parts of the plugin codebase.

Reply:
In that case I would prefer to have protected methods and a test double that expose these by making them public.

Reply:
@atimmer @jip @herregroen Can one of you give an opinion on this?

Reply:
Discussed this with @moorscode and we prefer the test double way. This is the way we've been using for other situations as well.

Reply:
@andizer Agreed. Instead of juggling around with other hacky approaches for this, I made the methods public, which doesn't hurt IMO. This way they can easily be tested.